dev/source/docs/ek3-affinity-lane-switching.rst

```diff
@@ -0,0 +1,68 @@
+
+
+**A simple example of an affinity configuration for a vehicle which has 1 Barometer, 2 GPS, 2 Airspeeds, 3 Magnetometers and 3 IMUs -**
+
+.. raw:: html
+
+   <table border="1" class="docutils">
+   <tr><th>LANE</th><th>1</th><th>2</th><th>3</th></tr>
+   <tr><td>AIRSPEED</td><td>1</td><td>2</td><td>1</td></tr>
+   <tr><td>BAROMETER</td><td>1</td><td>1</td><td>1</td></tr>
+   <tr><td>GPS</td><td>1</td><td>2</td><td>1</td></tr>
+   <tr><td>MAGNETOMETER</td><td>1</td><td>2</td><td>3</td></tr>
+   </table>
+
+*Numbers are the respective sensor instances*
+
+Configuration Parameters
+------------------------
+
+.. note::
+
+    Affinity is only available with EKF3, so make sure you are using it by ensuring :ref:`EK3_ENABLE <EK3_ENABLE>` is set to "1" and :ref:`AHRS_EKF_TYPE <AHRS_EKF_TYPE>` is set to "3"
+
+The :ref:`EK3_AFFINITY <EK3_AFFINITY>` parameter is a bitmask which gives you the option to choose the sensors you want to enable affinity for. Non-enabled will follow the default primary sensor allotment.
+
+The :ref:`EK3_ERR_THRESH <EK3_ERR_THRESH>` parameter controls the sensitivity of lane switching. Lane errors are accumulated over time relative to the active primary lane. This threshold controls how much of an error difference between a non-primary and primary lane is required to consider the former performing better. Lowering this parameter makes lane switching more responsive to smaller 'relative' errors, and in practical you will see a more aggressive lane switching, and, vice-versa. 
+
+.. warning::
+    
+    Misconfiguring the :ref:`EK3_ERR_THRESH <EK3_ERR_THRESH>` parameter could adversely affect the lane switching mechanism and have serious consequences which could lead to the loss of your vehicle. Please tune carefully.
+
+
+Test Results
+------------
+
+Following graphs are from SITL testing that show Affinity enabled lane changing when the primary lane's sensor is subjected to noise/malfunctioning.
+
+AIRSPEED
+++++++++
+An example of lane switching for a plane with 2 airspeed sensors and airspeed affinity enabled. There are 2 IMUs, hence 2 active lanes. The primary lane's airspeed sensor is failed to no change in pressure, hence reporting a constant value. The speed of the plane is increased and a lane switch occurs. Similarly, the second airspeed sensor of the second lane (now the primary lane) is failed and the plane's speed is decreased which again triggers a lane switch.
+
+.. image:: ../../../images/airspeed_affinity.png
+
+BAROMETER
++++++++++
+An example of lane switching for a plane with 2 barometers and barometer affinity enabled. There are 2 IMUs, hence 2 active lanes. The primary lane's barometer is failed to no change in pressure, hence reporting a constant value. The altitude of the plane is increased and a lane switch occurs. Similarly, the second baromter of the second lane (now the primary lane) is failed and the plane's altitude is decreased which again triggers a lane switch.
```

Comment:
"..lane's barometer has failed to show changes in pressure,"

spelling: barometer

has failed  instead of is failed


same changes under AIRSPEED

Reply:
corrected, thanks !